src/algo/depth-to-rgb-calibration/optimizer.cpp

```diff
@@ -435,7 +435,118 @@ void grid_xy(
         }
     }
 }
+template<class T>
+std::vector<double> interpolation(std::vector<T> const &grid_points, std::vector<double> x[], std::vector<double> y[], int dim, int valid_size, int valid_width)
```

Comment:
If possible, I'd like you to do what we did with valid-scene.cpp: put all the smearing-specific code in preprocess.cpp